Nodejs/Product/Analysis/Analysis/Analyzer/AnalysisUnit.cs

```diff
@@ -188,9 +188,9 @@ internal AnalysisUnit(JsAst ast, EnvironmentRecord environment)
         }
 
         internal virtual void AnalyzeWorker(DDG ddg, CancellationToken cancel) {
-            if (Tree != ProjectEntry.Tree) {
-                // we were enqueued and a new version became available, don't re-analyze against
-                // the old version.
+            if (Ast != null && ProjectEntry != null && ProjectEntry.Tree != null && Tree != ProjectEntry.Tree) {
```

Comment:
Okay... just reviewed this again with my alter ego from the future, and realized I was being an idiot. I intended to null check and *not* to flow through, and I totally misinterpreted your comment. fixed :smiley: